src/WebJobs.Script/Description/FSharp/FSharpCompiler.cs

```diff
@@ -45,30 +43,36 @@ public ICompilation GetFunctionCompilation(FunctionMetadata functionMetadata)
             bool debug = true;
 
             // First use the C# compiler to resolve references, to get consistenct with the C# Azure Functions programming model
-            Script<object> script = Microsoft.CodeAnalysis.CSharp.Scripting.CSharpScript.Create("using System;", options: _metadataResolver.CreateScriptOptions(), assemblyLoader: AssemblyLoader.Value);
+            Script<object> script = CodeAnalysis.CSharp.Scripting.CSharpScript.Create("using System;", options: _metadataResolver.CreateScriptOptions(), assemblyLoader: AssemblyLoader.Value);
             Compilation compilation = script.GetCompilation();
 
             var compiler = new SimpleSourceCodeServices();
 
-            // We currently create the script file in the directory itself
-            var scriptFile = Path.Combine(Path.GetDirectoryName(functionMetadata.ScriptFile),Path.ChangeExtension(Path.GetTempFileName(), "fsx"));
             FSharpErrorInfo[] errors = null;
-
-            //var exitCode = result.Item2;
             FSharpOption<Assembly> assemblyOption = null;
+            string scriptFilePath = Path.Combine(Path.GetTempPath(), Path.GetFileName(functionMetadata.ScriptFile));
+
             try
             {
+                var scriptFileBuilder = new StringBuilder();
+
                 // Write an adjusted version of the script file, prefixing some 'open' decarations
-                foreach (var import in script.Options.Imports)
+                foreach (string import in script.Options.Imports)
                 {
-                    File.AppendAllLines(scriptFile, new string[] { "open " + import });
-                    
+                    scriptFileBuilder.AppendLine("open " + import);
                 }
-                File.AppendAllLines(scriptFile, new string[] { "# 0 @\"" + functionMetadata.ScriptFile + "\"" });
 
+                // Suppress undesirable warnings
+                scriptFileBuilder.AppendLine("#nowarn \"988\"");
```

Comment:
Instead of filtering out post-compilation, I wanted to pass the suppression to the compiler. Unfortunately, `--nowarn` as a compiler option didn't seem to work when running in this mode, so I used the directive instead.